### ISSUE TITLE
MinGW Autotools & CMake builds are broken

### ISSUE THREAD

**Issue by lennoxho** [gperftools/gperftools#1486]
Autools MSYS [link](https://github.com/msys2/msys2-installer/releases/tag/2024-01-13) build:
```
g++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I./src  -I./src -DNO_TCMALLOC_SAMPLES  -pthread -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -Wno-unused-result -fsized-deallocation -faligned-new  -fno-omit-frame-pointer -momit-leaf-frame-pointer -g -O2 -MT src/tests/page_heap_test-page_heap_test.o -MD -MP -MF src/tests/.deps/page_heap_test-page_heap_test.Tpo -c -o src/tests/page_heap_test-page_heap_test.o `test -f 'src/tests/page_heap_test.cc' || echo './'`src/tests/page_heap_test.cc
In file included from D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/x86_64-w64-mingw32/bits/gthr-default.h:35,
                 from D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/x86_64-w64-mingw32/bits/gthr.h:148,
                 from D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/ext/atomicity.h:35,
                 from D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/bits/shared_ptr_base.h:61,
                 from D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/bits/shared_ptr.h:53,
                 from D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/memory:80,
                 from src/tests/page_heap_test.cc:13:
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/pthread.h:187:18: error: conflicting declaration 'typedef unsigned int pthread_key_t'
  187 | typedef unsigned pthread_key_t;
      |                  ^~~~~~~~~~~~~
In file included from ./src/windows/mingw.h:68,
                 from ./src/config.h:275,
                 from ./src/config_for_unittests.h:58,
                 from src/tests/page_heap_test.cc:8:
./src/windows/port.h:113:15: note: previous declaration as 'typedef DWORD pthread_key_t'
  113 | typedef DWORD pthread_key_t;
      |               ^~~~~~~~~~~~~
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/pthread.h:201:19: error: conflicting declaration 'typedef uintptr_t pthread_t'
  201 | typedef uintptr_t pthread_t;
      |                   ^~~~~~~~~
./src/windows/port.h:112:15: note: previous declaration as 'typedef DWORD pthread_t'
  112 | typedef DWORD pthread_t;
      |               ^~~~~~~~~
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/pthread.h:240:20: error: ambiguating new declaration of 'int sched_yield()'
  240 | int WINPTHREAD_API sched_yield(void);
      |                    ^~~~~~~~~~~
./src/windows/port.h:160:13: note: old declaration 'void sched_yield()'
  160 | inline void sched_yield(void) {
      |             ^~~~~~~~~~~
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/pthread.h:294:26: error: conflicting declaration of 'pthread_t pthread_self()' with 'C' linkage
  294 | pthread_t WINPTHREAD_API pthread_self(void);
      |                          ^~~~~~~~~~~~
./src/windows/port.h:115:18: note: previous declaration with 'C++' linkage
  115 | inline pthread_t pthread_self(void) {
      |                  ^~~~~~~~~~~~
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/pthread.h:297:26: error: ambiguating new declaration of 'int pthread_equal(pthread_t, pthread_t)'
  297 | int       WINPTHREAD_API pthread_equal(pthread_t t1, pthread_t t2);
      |                          ^~~~~~~~~~~~~
./src/windows/port.h:120:13: note: old declaration 'bool pthread_equal(pthread_t, pthread_t)'
  120 | inline bool pthread_equal(pthread_t left, pthread_t right) {
      |             ^~~~~~~~~~~~~
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/x86_64-w64-mingw32/bits/gthr-default.h: In function 'int __gthread_yield()':
D:/Users/LennoxHo/Downloads/New/msys64/ucrt64/include/c++/13.2.0/x86_64-w64-mingw32/bits/gthr-default.h:693:32: error: void value not ignored as it ought to be
  693 |   return __gthrw_(sched_yield) ();
      |                                ^
make: *** [Makefile:5866: src/tests/page_heap_test-page_heap_test.o] Error 1

```

CMake MinGW [link](https://github.com/brechtsanders/winlibs_mingw/releases/tag/13.2.0posix-17.0.5-11.0.1-ucrt-r3) build:
```
[  1%] Building CXX object CMakeFiles/logging.dir/src/base/logging.cc.obj
In file included from D:\Users\LennoxHo\source\gperftools\src\base\logging.cc:34:
D:/Users/LennoxHo/source/gperftools/build/config.h:6:10: fatal error: ../config.h: No such file or directory
    6 | #include "../config.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
mingw32-make[2]: *** [CMakeFiles\logging.dir\build.make:76: CMakeFiles/logging.dir/src/base/logging.cc.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:890: CMakeFiles/logging.dir/all] Error 2
mingw32-make: *** [makefile:165: all] Error 2
````

PR https://github.com/gperftools/gperftools/pull/1485 should fix both builds.

**Comment by lennoxho** [comment]
Will be addressed as described here https://github.com/gperftools/gperftools/issues/1483#issuecomment-1894500594

**Comment by lennoxho** [comment]
Autotools build fixed with #1487 
CMake build will be addressed in another PR
